test/e2e/generate_incident.go

```diff
@@ -0,0 +1,201 @@
+package osde2etests
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+	"time"
+
+	"github.com/google/uuid"
+)
+
+type PagerDutyClient interface {
```

Comment:
To note that this might get confusing, as we already have a pd client in the main code. Maybe calling it `TestPagerDutyClient` would offer more clarity.